src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1624,9 +1624,7 @@ public IndexDefinition GetIndexDefinition()
                         // we have to open read tx for mapResults _after_ we open index tx
 
                         if (query.WaitForNonStaleResultsAsOfNow && query.CutoffEtag == null)
-                            query.CutoffEtag =
-                                Collections.Max(
-                                    x => DocumentDatabase.DocumentsStorage.GetLastDocumentEtag(documentsContext, x));
+                            query.CutoffEtag = Collections.Max(x => GetLastDocumentEtagInCollection(documentsContext, x));
```

Comment:
This is likely to be a hotspot, can we do without the lambda here? It allocates and does indirect calls.
Better to use a `foreach`